Version 1.2.0 open for comments

CoralineAda opened the pull request.

sinthetix committed:
Added Healing Rain to projects list

doodzik committed:
add Bundler to list of Projects Using the Contributor Covenant

CoralineAda committed:
Merge pull request #27 from doodzik/master

add Bundler to list of Projects Using the Contributor Covenant

doodzik committed:
fix link to Contributor Covenant

CoralineAda committed:
Merge pull request #28 from doodzik/master

fix link to Contributor Covenant

bruceadams committed:
Add pmap gem

`pmap` is a pretty small Ruby Gem. I'm not sure it really deserves to be on this list. In any case, I'm happy to have added this code of conduct to `pmap`.

jvanbaarsen committed:
GitLab also supports the Code of Conduct

Signed-off-by: Jeroen van Baarsen <jeroenvanbaarsen@gmail.com>

CoralineAda committed:
Merge pull request #30 from jvanbaarsen/added-gitlab

GitLab also supports the Code of Conduct

CoralineAda committed:
Merge pull request #29 from bruceadams/patch-2

Add pmap gem

Commit:
add opendronemap to list of projects

CoralineAda committed:
Merge pull request #31 from smathermather/patch-2

add opendronemap to list of projects

moonglum committed:
Added Hacken.in to the list

Hacken.in just adopted the contributor covenant. You can find see [Code of Conduct](https://github.com/hacken-in/website/blob/master/CODE_OF_CONDUCT.md) here and a list of all people currently working on this project saying *aye* [here](https://github.com/hacken-in/website/pull/439). You can also find hints to find the file in both the [README.md](https://github.com/hacken-in/website/blob/master/README.md) and the [CONTRIBUTING.md](https://github.com/hacken-in/website/blob/master/CONTRIBUTING.md) files.

CoralineAda committed:
Merge pull request #32 from moonglum/patch-2

Added Hacken.in to the list

iancooper committed:
Added the Paramore project as a user

CoralineAda committed:
Merge pull request #33 from iancooper/master

Added the Paramore project as a user

Commit:
Update README.md

added haxe-pixi project

Commit:
Update README.md

added haxe-pixi project

CoralineAda committed:
Merge pull request #34 from adireddy/master

haxe-pixi

theunixman committed:
All haskellnow.org projects.

CoralineAda committed:
Merge pull request #35 from misandrist/master

All haskellnow.org projects.

valpackett committed:
Add HTTPotion to the project list

CoralineAda committed:
Merge pull request #36 from myfreeweb/patch-2

Add HTTPotion to the project list

jodosha committed:
Added Lotus

CoralineAda committed:
Merge pull request #37 from jodosha/lotus

Added Lotus

CoralineAda committed:
Add ethnicity

CoralineAda committed:
Update README.md

floriank committed:
Add OpenProject to the project list

CoralineAda committed:
Merge pull request #44 from floriank/master

Add OpenProject to the project list

snipe committed:
Added Snipe-IT

CoralineAda committed:
Merge pull request #45 from snipe/master

Added Snipe-IT

orta committed:
Adds CocoaPods + Order by Alphabetical

orta committed:
Re-add Snipe-IT - sorry!

CoralineAda committed:
Merge pull request #46 from orta/master

Adds CocoaPods + Order by Alphabetical

gcaprio committed:
Adding ED Project

CoralineAda committed:
Merge pull request #47 from eldest-daughter/master

Adding Eldest Daughter Questionnaire Project

CoralineAda committed:
Added ruby-community

CoralineAda committed:
Merge branch 'master' of github.com:CoralineAda/contributor_covenant

vassilevsky committed:
Make source information italic

The code looks a little bit better this way

CoralineAda committed:
Merge pull request #49 from vassilevsky/italic-source

Make source information italic

Commit:
add: two Haskell projects

* Fill in the Blanks
* wai-request-spec

CoralineAda committed:
Merge pull request #50 from cabrera/hprojs

add: two Haskell projects

CoralineAda committed:
Update README.md

Fix sort order

havenwood committed:
Add RVM

CoralineAda committed:
Merge pull request #51 from havenwood/patch-2

Add RVM

Commit:
Update README.md

CoralineAda committed:
Merge pull request #54 from DecipherCode/patch-2

Update README.md - Add project reference

rheinwein committed:
Add Panamax to list of projects

CoralineAda committed:
Merge pull request #55 from rheinwein/patch-2

Add Panamax to list of projects

DrMcCoy committed:
Add xoreos to list of projects

CoralineAda committed:
Merge pull request #58 from DrMcCoy/master

Add xoreos to list of projects

Commit:
Add Monsti CMS to projects.

johana-star committed:
Update README.md

The Contributor Covenant was merged into Opal in their [issue #947](https://github.com/opal/opal/pull/947).

Commit:
Update CODE_OF_CONDUCT.md to 1.1.0

solnic committed:
Add rom-rb to the list of projects

CoralineAda committed:
Copy to master

CoralineAda committed:
Merge pull request #61 from strand/patch-2

Update README.md

CoralineAda committed:
Merge pull request #62 from strand/update-to-1.1.0

Update CODE_OF_CONDUCT.md to 1.1.0

CoralineAda committed:
Merge pull request #63 from solnic/patch-2

Add rom-rb to the list of projects

CoralineAda committed:
Merge pull request #60 from chrneumann/add-monsti-project

Add Monsti CMS to projects.

CoralineAda committed:
Remove Opal

JPLeBreton committed:
Update README.md

added Playscii

CoralineAda committed:
Merge pull request #67 from JPLeBreton/patch-2

Update README.md

CoralineAda committed:
Fix URLs

CoralineAda committed:
Merge branch 'master' of github.com:CoralineAda/contributor_covenant

johana-star committed:
Fixed link to Contributor Covenant

graycat committed:
added PyBBIO project

CoralineAda committed:
Merge pull request #70 from strand/patch-2

Fixed link to Contributor Covenant

CoralineAda committed:
Merge pull request #71 from graycatlabs/master

added PyBBIO project

CoralineAda committed:
Merge pull request #72 from djbender/add-healing-rain

Added Healing Rain to projects list

johana-star committed:
Update README.md

CoralineAda committed:
Merge pull request #77 from strand/patch-3

Update README.md

okeuday committed:
Add CloudI to projects list

CoralineAda committed:
Merge pull request #81 from okeuday/add-CloudI-project

Add CloudI to projects list

CoralineAda committed:
version 1.2.0

CoralineAda committed:
Add nationality

webmaven commented on version/1/2/0/code_of_conduct.md:
```diff
@@ -0,0 +1,23 @@
+
+As contributors and maintainers of this project, we pledge to respect all people who contribute through reporting issues, posting feature requests, updating documentation, submitting pull requests or patches, and other activities.
+
+We are committed to making participation in this project a harassment-free experience for everyone, regardless of level of experience, gender, gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, religion, nationality, or political affiliation.
```
How about *"religion or lack thereof"*?

EndangeredMassa replied:
As an atheist, I assume simply saying "religion" also covers my lack thereof. However, I don't speak for others. I just wanted to offer that data point.

webmaven replied:
I've definitely observed conversations in various online fora where atheism is *not* so assumed. In fact, atheism is what all the fundamentalists can agree among themselves to gang up on as a shared mutual threat.

EndangeredMassa commented on version/1/2/0/code_of_conduct.md:
```diff
@@ -0,0 +1,23 @@
+
+Instances of abusive, harassing, or otherwise unacceptable behavior may be reported by opening an issue or contacting one or more of the project maintainers.
+
+This Code of Conduct is adapted from the [Contributor Covenant](http://contributor-covenant.org), version 1.2.0, available at [http://contributor-covenant.org/version/1/2/0/](http://contributor-covenant.org/version/1/2/0/)
```
Should this line end with a period?

postmodern replied:
Not usually when the sentence ends with a fully qualified URL. It makes it difficult for users to manually copy/paste the URL.

unixorn commented on version/1/2/0/code_of_conduct.md:
```diff
@@ -0,0 +1,23 @@
+* Personal attacks
+* Trolling or insulting comments
+* Public or private harassment
+* Publishing other's private information, such as physical or electronic addresses, without permission
```
Perhaps s/without permission/without their explicit permission/ to get around someone saying "Hey, I'm publishing everyone's email unless they say no."

Should probably also explicitly forbid publishing phone numbers too.

webmaven replied:
ObNitPick: *others'*

postmodern commented on version/1/2/0/code_of_conduct.md:
```diff
@@ -0,0 +1,23 @@
+
+Examples of unacceptable behavior by participants include:
+
+* The use of sexual language or imagery
```
Would this negatively impact projects that focus on Sex Education? Maybe use the word `sexualized` instead?

webmaven replied:
In another conversation, it was suggested that 'overtly' would be a good qualifier here.

postmodern replied:
I think "overt" would help differentiate between say tasteful nude artwork and adult content.

achadwick replied:
I work on an [art program](https://github.com/mypaint), and I'm interested in growing its community and keeping working on the code a non-sucky experience for everyone. This question is relevant to us in odd ways:

* Certain standard test images I will not use as a matter of principle, and I would expect contributors not to within our project spaces (ref. Wikipedia: "Lenna#Controversy", page contains a cropped part of a sexualized image above the referenced section, just the woman's head)
* We require screenshots of the program in operation, for publicity and for debugging glitches. Those obv. have to be SFW (and safe for kids), but threads designed to gather them may by accident not always be (artist-testers should not be not be bound by our CoC outside our project spaces). The practical way of handling an accidental overstep should be to replace with a link or delete, with an appropriate request to refrain or a warning. Fine where we have control of course...
* I want artist-testers to be able to draw whatever the hell they want in spaces coded for that sort of thing. Our CoC should expect them not to use that freedom to harass without consequence, but it should also accept that some non-project spaces are sex-positive, kink-friendly, provide sex ed., or are just artist hangouts where folks might talk about the neat programs they use or hack on in the same breath as talking about the art they make.

I vacillate on this so much. I do not like the word "overt" as a replacement. May I propose

> * Inappropriate sexual language or imagery. Within this project's spaces, sexual language or imagery of any sort is never appropriate.

as a replacement? The implication being that outside project spaces, interpretation hinges on what is appropriate - there. And that directed harassment as interpreted anywhere is never appropriate, full stop.

postmodern commented on version/1/2/0/code_of_conduct.md:
```diff
@@ -0,0 +1,23 @@
+* Trolling or insulting comments
+* Public or private harassment
+* Publishing other's private information, such as physical or electronic addresses, without permission
+* Other unethical or unprofessional conduct.
```
Unprofessional is ambiguous as there is no agreed upon definition of what is a professional OSS contributor/maintainer. Might also unfairly impact those new to OSS who do not understand the development process.

webmaven replied:
Also, 'ethics' usually refers to 'professional ethics', and well, we aren't a profession.

postmodern replied:
I think "unethical" should be expanded into specific violations (ex: stealing someone else's work and claiming it as your own).

sigmavirus24 replied:
> I think "unethical" should be expanded into specific violations (ex: stealing someone else's work and claiming it as your own).

I would caution against the temptation to start enumerating finite lists of violations. That way leads loopholes and corner cases people didn't think about.

postmodern replied:
Leaving "unethical" and "unprofessional" undefined also allows for loopholes and creative interpretation.

achadwick replied:
This list should probably have some "including but not limited to" phrasing if the goal is to leave it extensible, or open to the interpretation of the project admins.

Since the body of the list basically _defines_ what is not acceptable in a professional demeanour, I'd be happy with the final bullet point being struck out.

postmodern replied:
My only concern about "included but not limited to" is it would allow for selective interpretation.

webmaven replied:
Well, yeah project admins get flexibility for interpretation and enforcement against otherwise un-enumerated behaviors.

Otherwise you're giving the "everything not explicitly forbidden is allowed" crowd free rein by tying the admins' hands: "The CoC doesn't *say* I can't make a parody twitter account using their name that asks for money in return for various sexual acts, does it?"

postmodern replied:
> Well, yeah, project admins get flexibility for interpretation and enforcement against otherwise un-enumerated behaviors.

Isn't this the problem we're trying to solve? Historically, maintainers have not been good at identifying and punishing harassment; often even defending the harasser. Faux Codes of Conduct (ex: "be excellent to each other") have not worked due to flexible interpretation. By having an explicit Code of Conduct it removes any doubt or question about what's isn't acceptable and what course of action needs to be taken.

> Otherwise you're giving the "everything not explicitly forbidden is allowed" crowd free rein by tying the admins' hands

By leaving the Code of Conduct open ended we're also leaving a loophole for those that claim they are being persecuted due to "reverse sexism". Also, I'm pretty sure your hypothetical example violates Twitter's ToS, as it crosses the line between parody and defamation/libel.

webmaven replied:
@postmodern:
> Isn't this the problem we're trying to solve? Historically, maintainers have not been good at identifying and punishing harassment;

Which is why we're giving them examples. But examples are just that, trying to create a comprehensive list of forbidden behaviors would be an exercise in futility, and the document itself would be as unreadable as a typical EULA.

> we're also leaving a loophole for those that claim they are being persecuted due to "reverse sexism".

That isn't a loophole, they can and will claim that anyway. In fact, they'll probably claim that the mere existence of a CoC is itself "reverse sexism".

> Also, I'm pretty sure your hypothetical example violates Twitter's ToS, as it crosses the line between parody and defamation/libel.

Don't make me get really creative, that will just give the wrong people new ideas to use. :bomb:

postmodern replied:
Two more loopholes that came to mind:

1. Maintainers sympathetic with the harasser can argue that it's not really harassment and that's not how _they_ interpret the Code of Conduct.
2. Contributors/maintainers who have a grudge against another contributor/maintainer waiting for them to slip up, and arguing that their minor transgression violates the Code of Conduct and should be banned.

I would like to see these loopholes closed. I disagree that enumerating behavior or clarifying language is futile, as that is what we've been doing in the issues so far and it's yielded results.

webmaven replied:
> I would like to see these loopholes closed.

That would be nice, but there will *always* be room for some interpretation, now matter how hard you try to make fuzzy lines bright and hard.

> I disagree that enumerating behavior or clarifying language is futile

Okay, I don't think adding more examples or clarifying language is futile, only that trying to make the list fully comprehensive is. After a while enumerating more hits diminishing returns, after which you will still need "Including but not limited to" language.

We may have reached the point where you and I will have to agree to disagree.

postmodern commented on version/1/2/0/code_of_conduct.md:
```diff
@@ -0,0 +1,23 @@
+* Publishing other's private information, such as physical or electronic addresses, without permission
+* Other unethical or unprofessional conduct.
+
+Project maintainers have the right and responsibility to remove, edit, or reject comments, commits, code, wiki edits, issues, and other contributions that are not aligned to this Code of Conduct. Project maintainers who do not follow the Code of Conduct may be removed from the project team.
```
`removed from the project team` needs clarification. This might allow being moved from the core team to a less visible team. Being "removed" from the project, but still hanging around the IRC channel, forum, mailing list. Would they still be allowed to contribute as a regular contributor?

webmaven replied:
I'd say, that this is up to the community in question.

CoralineAda committed:
Added ncolony

CoralineAda committed:
Merge branch 'master' of github.com:CoralineAda/contributor_covenant

CoralineAda committed:
Add oapi

alto committed:
add AASM to list of projects

CoralineAda committed:
Merge pull request #88 from aasm/add-aasm

add AASM to list of projects

rtorr committed:
adding https://github.com/rtorr/vim-cheat-sheet to the list of projects using this code of conduct

CoralineAda committed:
Merge pull request #91 from rtorr/patch-2

adding https://github.com/rtorr/vim-cheat-sheet

webmaven commented on version/1/2/0/code_of_conduct.md:
```diff
@@ -0,0 +1,23 @@
+
+We are committed to making participation in this project a harassment-free experience for everyone, regardless of level of experience, gender, gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, religion, nationality, or political affiliation.
+
+Examples of unacceptable behavior by participants include:
```
Perhaps this small tweak?:

> Some examples of unacceptable behavior by participants include, but are not limited to:

postmodern replied:
I worried that leaving it open ended would allow for abuse. It should be possible to enumerate the various forms of harassment?

unixorn replied:
It should be, but it isn't. People are creative and will come up with new ways to abuse each other.

webmaven commented:
@karenetheridge, good suggestion. :+1:

Commit:
Add Composer to list of projects

It mentions the Contributor Covenant here: https://github.com/composer/composer/blob/master/CONTRIBUTING.md

CoralineAda committed:
Merge pull request #92 from jrobeson/patch-2

Add Composer to list of projects

Commit:
Adding QA-Tools to the list of projects

CoralineAda committed:
Merge pull request #93 from aik099/patch-2

Adding QA-Tools to the list of projects

julianguyen committed:
Added if me to project links

CoralineAda committed:
Merge pull request #94 from julianguyen/master

Added if me to project links

ktheory committed:
Update README.md: add Rack::Attack project

CoralineAda committed:
Merge pull request #95 from ktheory/patch-2

Update README.md: add Rack::Attack project

sebmck committed:
Add Babel to examples of projects

CoralineAda committed:
Merge pull request #96 from sebmck/sebmck-patch-1

Add Babel to list of projects

CoralineAda committed:
Add Rubygems.org

bboe committed:
Add PRAW.

CoralineAda committed:
Merge pull request #97 from bboe/patch-2

Add PRAW.

benlesh committed:
add ReactiveX/RxJS

CoralineAda committed:
Merge pull request #98 from blesh/master

add ReactiveX/RxJS

Commit:
Add angular-formly

Thank you for creating this.

CoralineAda committed:
Merge pull request #99 from kentcdodds/patch-2

Add angular-formly

CoralineAda committed:
Additional strengthening language changes.

CoralineAda committed:
Updated project list

CoralineAda committed:
Merge remote-tracking branch 'origin' into 1_2_0

CoralineAda committed:
Language tweaks

DvalCaruso reviewed (commented):
cleaning up

kaungkin499 commented:
[kaungkincom (1).csv](https://github.com/EthicalSource/contributor_covenant/files/7447283/kaungkincom.1.csv)

Ericsandanu reviewed (commented):
Duplicate of #

- [ ] #
Duplicate of #
Duplicate of #CODE_OF_CONDUCT.md

1. [ ]  

- [ ] [##[](url)]([url](url))
